File: sem3 pdfs/dsacodes/week_2/binarysearch.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
	int m,b,e,n,sear,arr[100];
	cout<<"enter number of elements";
	cin>>n;
	cout<<"enter elements";
	int i;
	for(i=0;i<n;i++)
		cin>>arr[i];
	cout<<"enter searched element";
	cin>>sear;
	b=0;
	e=n-1;
	
	while(b<=e)
	{

		m=(b+e)/2;
		if(arr[m]==sear)
		{
	
			cout<<"element "<<sear<<" is found at "<<m<<"th index";
			break;
		}
		else if(arr[m]>sear)
		{
			
			e=m-1;
		}
		else if(arr[m]<sear)
		{
			
			b=m+1;
		}
	
		
		
	}
	if(b>e)
	{
		
		cout<<"element not found";
	}
}

File: sem4 pdfs/DAA lab/WEEK_2/Binarysearch.java
import java.util.Scanner;
class Binarysearch{
    public static void main(String args[])
    {
        Scanner sc=new Scanner(System.in);
        System.out.println("Enter size of array");
        int n=sc.nextInt();
        System.out.println("Enter search element");
        int sear=sc.nextInt();
        int arr[]=new int[n];
        int s=0,e=n-1,m,i,flag;
        System.out.println("Enter array elements");
        for(i=0;i<n;i++){
            arr[i]=sc.nextInt();
        }
        m=(s+e)/2;
        while(s<=e){
            if(arr[m]==sear){
                System.out.println("element found at index "+m);
                break;

            }
            else if(sear>arr[m])
            {
                s=m+1;
            }
            else if(sear<arr[m]){
                e=m-1;
            }
            m=(s+e)/2;
        }
        if(s>e){
            System.out.println("Search element not found");
        }
        
    }
}